packages/eslint-plugin/src/RuleModule.ts

```diff
@@ -0,0 +1,404 @@
+import { TSESTree } from '@typescript-eslint/typescript-estree';
+import { ParserServices } from '@typescript-eslint/parser';
+import { AST, Linter, Scope } from 'eslint';
+import { Comment as ESTreeComment } from 'estree';
```

Comment:
you should use `TSESTree.Comment` here

Reply:
Probably should, yeah.
I should add those interfaces properly into the TSESTree types.
Ideally we should be able to uninstall estree at the end of this.